quil-rs/src/instruction/gate.rs

```diff
@@ -128,13 +131,53 @@ impl fmt::Display for GateModifier {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, strum::Display, strum::EnumString)]
+#[strum(serialize_all = "UPPERCASE")]
+pub enum PauliWord {
+    I,
+    X,
+    Y,
+    Z,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PauliTerm {
+    pub words: Vec<PauliWord>,
+    pub expression: Expression,
+    pub arguments: Vec<String>,
```

Comment:
Similar to https://github.com/rigetti/quil-rs/pull/177/files#r1146660689, should/could these be made private so that using `PauliTerm::new` is required?